src/lib/server/endpoints/anthropic/endpointAnthropic.ts

```diff
@@ -0,0 +1,89 @@
+import { z } from "zod";
+import { buildPrompt } from "$lib/buildPrompt";
+import { ANTHROPIC_API_KEY } from "$env/static/private";
+import type { Endpoint } from "../endpoints";
+import type { TextGenerationStreamOutput } from "@huggingface/inference";
+
+export const endpointAnthropicParametersSchema = z.object({
+	weight: z.number().int().positive().default(1),
+	model: z.any(),
+	type: z.literal("anthropic"),
+	baseURL: z.string().url().default("https://api.anthropic.com"),
+	apiKey: z.string().default(ANTHROPIC_API_KEY ?? "sk-"),
+	defaultHeaders: z.record(z.string()).optional(),
+	defaultQuery: z.record(z.string()).optional(),
+});
+
+export async function endpointAnthropic(
+	input: z.input<typeof endpointAnthropicParametersSchema>
+): Promise<Endpoint> {
+	const { baseURL, apiKey, model, defaultHeaders, defaultQuery } =
+		endpointAnthropicParametersSchema.parse(input);
+	let Anthropic;
+	try {
+		Anthropic = (await import("@anthropic-ai/sdk")).default;
+	} catch (e) {
+		throw new Error("Failed to import OpenAI", { cause: e });
+	}
+
+	const anthropic = new Anthropic({
+		apiKey,
+		baseURL,
+		defaultHeaders,
+		defaultQuery,
+	});
+
+	return async ({ messages, preprompt, continueMessage }) => {
+		const prompt = await buildPrompt({
+			messages,
+			continueMessage,
+			model,
+		});
+
+		let tokenId = 0;
+		return (async function* () {
+			const stream = anthropic.messages.stream({
+				model: model.id ?? model.name,
+				messages: [{ role: "user", content: prompt }],
+				max_tokens: model.parameters?.max_new_tokens,
+				temperature: model.parameters?.temperature,
+				top_p: model.parameters?.top_p,
+				top_k: model.parameters?.top_k,
+				stop_sequences: model.parameters?.stop,
+				// TODO: is it correct to provide the preprompt here since the client
+				// natively supports it? or in the buildPrompt function?
+				system: preprompt,
+			});
```

Comment:
```suggestion
	return async ({ messages, preprompt }) => {
		let system = preprompt;
		if (messages?.[0]?.from === "system") {
			system = messages[0].content;
		}

		const messagesFormatted = messages
			.filter((message) => message.from !== "system")
			.map((message) => ({
				role: message.from,
				content: message.content,
			})) as unknown as {
			role: "user" | "assistant";
			content: string;
		}[];

		let tokenId = 0;
		return (async function* () {
			const stream = anthropic.messages.stream({
				model: model.id ?? model.name,
				messages: messagesFormatted,
				max_tokens: model.parameters?.max_new_tokens,
				temperature: model.parameters?.temperature,
				top_p: model.parameters?.top_p,
				top_k: model.parameters?.top_k,
				stop_sequences: model.parameters?.stop,
				system,
			});
```

			
			
This should do the trick. `buildPrompt` is not needed in this case, it's used for converting an array of message to a formatted string for APIs that don't have a messages interface, but in the case of anthropic looks like you can just pass an array of message directly. So we format it correctly (they don't support `role: "system"`) and pass the system prompt separately in `system`